src/txmempool.cpp

```diff
@@ -410,7 +410,8 @@ void CTxMemPool::removeUnchecked(txiter it, MemPoolRemovalReason reason)
         // for any reason except being included in a block. Clients interested
         // in transactions included in blocks can subscribe to the BlockConnected
         // notification.
-        GetMainSignals().TransactionRemovedFromMempool(it->GetSharedTx());
+        GetMainSignals().TransactionRemovedFromMempool(it->GetSharedTx(), reason == MemPoolRemovalReason::CONFLICT
```

Comment:
See above, on the practical-side I wanted to avoid linking mempool code in wallet. But we can overcome this, so real question, above fixing behavior, is if we pass `EXPIRY` or `SIZELIMIT` to the wallet, does it going to adapt its broadcast on this or take any other action ?

Right now I think the wallet doesn't have any code to do this, so I think the discussion is worthwhile but beyond this PR ?

Actually, digging into this,  `MemPoolRemovalReason` purpose seems weak right now because we only use it in `removeUnchecked` to decide if we should trigger `TransactionsRemovedFromMempool` ?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/18600#discussion_r407944782

> Actually, digging into this, `MemPoolRemovalReason` purpose seems weak right now because we only use it in `removeUnchecked` to decide if we should trigger `TransactionsRemovedFromMempool` ?

This thread was continued https://github.com/bitcoin/bitcoin/pull/18600#discussion_r407550301 where Marco made a similar suggestion. But to respond to "purpose seems weak", the suggestion wasn't based on what wallet conflict tracking code is currently doing. Wallet conflict tracking has changed before and will likely change in the future. The suggestion to use the existing reason code was avoid to being inconsistent and confusing and avoid implementing wallet behavior outside of the wallet